### ISSUE TITLE
Diagnostics not linking the code properly

### ISSUE THREAD

**Issue by lambda-0x** [dojoengine/dojo#2027]
**Describe the bug**
> $ sozo build

```
warn[E0001]: Unused variable. Consider ignoring by prefixing with `_`.
 --> /path/to/dojo/examples/spawn-and-move/src/actions.cairo:16:1
#[dojo::contract]
^***************^
```

**To Reproduce**
simply add a unused variable in the code and build

**Expected behavior**
should point to the correct line where issue occured.

**Comment by lambda-0x** [comment]
its working as expected until `0.7.0-alpha.3` and i am seeing this behaviour from `0.7.0-alpha.4` onwards.

**Comment by lambda-0x** [comment]
did a `git bisect` on it and the first commit that has the issue is `e6146516d91fe5e91dc6ca1259b8699e40828e40`

**Comment by glihm** [comment]
> did a `git bisect` on it and the first commit that has the issue is `e6146516d91fe5e91dc6ca1259b8699e40828e40`

And to complete this, it is only happening on the `#[dojo::contract]` attribute. Others like `#[dojo::model]`, `#[dojo::interface]` and `#[dojo::event]` are working as expected.

**Comment by glihm** [comment]
Seems like it is the case with all attributes. It is being investigated by Scarb team.

**Comment by glihm** [comment]
Closed in #2141.
